### PR TITLE
Movies and Liked Components

### PR DESCRIPTION
**Pull request by tejas-muthya37**
This PR contains the code for the Movies and Liked Components.

The code to be reviewed is present in src -> Screens -> Movies, src -> Screens -> Liked and src -> Components -> CategoryMovies.

### CODE REVIEW

File: src/Components/CategoryMovies/CategoryMovies.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import Movie from "./../Movie/Movie";
+import { useProducts } from "./../../Context/products-context.js";
+import { useVideos } from "./../../Context/videos-context";
+import { useEffect } from "react";
+
+const CategoryMovies = ({ category }) => {
+  const { stateVideo, dispatchVideo } = useVideos();
+
+  useEffect(() => {
+    fetch("/api/videos", {
```
Comment (SJTGSHIVAM) [inline src/Components/CategoryMovies/CategoryMovies.js:11]: I would recommend you to use axios it makes are life a lot easier.

File: src/Screens/Liked/liked.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    flex-wrap: wrap;
+    justify-content: space-around;
+    gap: 4rem;
+}
+
+.liked-videos-section>*:not(h1) {
+    width: 17.5rem;
+}
+
+.Liked h1 {
```
Comment (SJTGSHIVAM) [inline src/Screens/Liked/liked.css:19]: never saw anyone using capital naming in CSS. Just keep a check on whats the usual practice.

File: src/Screens/Liked/Liked.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    localStorage.setItem("DISLIKED_ARRAY", JSON.stringify(state.dislikedArray));
+  }, [
+    state.likedArray,
+    state.watchLaterArray,
+    state.historyArray,
+    state.dislikedArray,
+  ]);
+  return (
+    <div className="Liked">
+      {state.likedArray.length > 0 && <h1>Liked videos</h1>}
```
Comment (SJTGSHIVAM) [inline src/Screens/Liked/Liked.js:41]: Try to destructure the state so its easier to write.

**Review by SJTGSHIVAM — APPROVED** [review]
Things look fine.

File: src/Components/CategoryMovies/CategoryMovies.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import "./categoryMovies.css";
+import Movie from "./../Movie/Movie";
+import { useProducts } from "./../../Context/products-context.js";
+import { useVideos } from "./../../Context/videos-context";
```
Comment (yaswanthmyneni) [inline src/Components/CategoryMovies/CategoryMovies.js:4]: suggestion: create index file in context folder and import all context from it

File: src/Components/CategoryMovies/CategoryMovies.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            movie.category === category && (
+              <Movie
+                key={movie._id}
+                _id={movie._id}
+                source={movie.source}
+                thumbnail={movie.thumbnail}
+                title={movie.title}
+                category={movie.category}
+                addToLiked={() =>
+                  dispatch({ type: "Add to Liked", payload: movie })
```
Comment (yaswanthmyneni) [inline src/Components/CategoryMovies/CategoryMovies.js:52]: suggestion: send movie like < Movie key={movie._id} movie={movie} /> and destructure in < Movie > component.

File: src/Context/products-context.js [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+  fetch("/api/user/history/all", {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
```
Comment (yaswanthmyneni) [inline src/Context/products-context.js:104]: changes: you can remove console logging data when you push code to the GitHub

**Review by yaswanthmyneni — APPROVED** [review]
everything looks good to merge

File: src/Components/CategoryMovies/CategoryMovies.js (hunk 1 above)
Comment (yaswanthmyneni) [reply src/Components/CategoryMovies/CategoryMovies.js:11]: yeah, try Axios for API calls.

**Pull request merged**